internal/worker.go

```diff
@@ -146,6 +146,11 @@ type (
 		// default: false
 		DisableActivityWorker bool
 
+		// EnableActivityLocalDispatch allows workflow worker to dispatch activity tasks through local tunnel after decisions are made.
+		// This is an performance optimization to skip activity scheduling efforts.
+		// If it is enabled then DisableActivityWorker option must be false.
+		EnableActivityLocalDispatch bool
```

Comment:
If I understand the code correctly, the same parameter will be used for both normal and local dispatch activity workers, then enabling this feature will double the # of activity workers (# of goroutines) on user worker hosts. 

Those added goroutines will also compete with the normal activity worker. If user worker is already under high load, enabling this feature may result in even longer activity execution time as there will be more concurrent activities on the host. I am not sure if this will be a concern or not.

Reply:
Yes it will double the # of goroutines.
Ideally we should just combine the poller and leave the processor (and its go-routines intact). This was discussed below and it requires some more refactoring not specific to this task. So we decided to postpone to a later PR.
# of goroutines might not be a big issue for worker. Regardless there might be other resource constraints and we need to tune the parameters to find a good default value.